projects/dsp-ui/src/lib/action/components/login-form/login-form.component.spec.ts

```diff
@@ -0,0 +1,205 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { FormBuilder, FormsModule, ReactiveFormsModule} from '@angular/forms';
+import { MatInputModule } from '@angular/material/input';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import {
+    ApiResponseData,
+    AuthenticationEndpointV2,
+    LoginResponse,
+    LogoutResponse,
+    MockUsers,
+    UsersEndpointAdmin
+} from '@dasch-swiss/dsp-js';
+import { of } from 'rxjs';
+import { AjaxResponse } from 'rxjs/ajax';
+import { DspApiConnectionToken, Session, SessionService } from '../../../core';
+import { LoginFormComponent } from './login-form.component';
+
+/**
+ * Test host component to simulate login-form component.
+ */
+@Component({
+    template: `<dsp-login-form #loginForm></dsp-login-form>`
```

Comment:
In the test host component, you should register to the @Output status. Then you can trigger the emission and see if it emits the correct status.

Reply:
like in:
https://github.com/dasch-swiss/dsp-ui-lib/blob/b3b1fee1ecc97085f3bdb199c0352c36825b206a/projects/dsp-ui/src/lib/search/advanced-search/select-ontology/select-ontology.component.spec.ts#L14-L44

Reply:
I attempted this in 5ff6925. However, in the callback, the status is emitted within a setTimeout function which makes it so that my test fails because it isn't emitted in time for the assertion.

@kilchenmann does this need to be in a setTimeout function?
https://github.com/dasch-swiss/dsp-ui-lib/blob/5ff6925f133de3b7ccb027fa10c8739c7b5623ac/projects/dsp-ui/src/lib/action/components/login-form/login-form.component.ts#L139-L142

Reply:
refactored in 32e144c